Feat/creat landing-page and navbar logged in

jaredsina opened the pull request:
Updates: 

- Generalize navbar to be used in App.jsx
- Update Navbar component to display different information for logged in vs logged out users
- Minor error fix on Hero component
- Update LandingPage component with new view for logged in users
- Update Hero component to display book images from the database
- Create UserActivity Component for the logged in users landing page
- Create UserReviews Component for the logged in users landing page


How to review:

- [ ] View Navbar Component Logged In vs Logged Out
  - Navbar component has an isAuthenticated variable that can be set to true or false to see both versions of the navbar.

- [ ] View Landing Page Logged In vs Logged Out
  - LandingPage has an isAuthenticated variable that can be set to true or false to see both versions of the component

- [ ] LandingPage book images are rendered

- [ ] Using chrome tools view mobile versions of pages for correct sizing

notes: 
- Need to add the authenticated logic to conditional render information and for the logout button
- Need to add react-router configuration to the buttons to other pages once they are created
- Need to update features images to include screenshots of the finished app

alexh205 reviewed (commented):
Overall everything looks great! 

I did notice a few things, which I have highlighted below:

- In the NavBar component [Logged In user]: The drop-down menu should automatically close when a click is triggered outside of the menu. See gif below for the current effect when the screen is manually resized.

![Moving DropDown Menu](https://github.com/chingu-voyages/v49-tier3-team-29/assets/95322089/1aacf654-f4c1-4439-bfe5-edb8ca6d823e)

- The number of displayed book objects should decrease with the screen media size (5 books ->3 books -> 2 books ).

jackie-leary commented on client/src/App.jsx:
```diff
@@ -18,22 +25,31 @@ function App() {
+						</Route>
 						<Route
-							path='/login'
+							path="/login"
```
For some reason I'm redirected to the landing page when I try to navigate to /login. 🤔 Every other existing route works though.

jaredsina replied:
You are correct, some changes were made to the login page component that seems to have made it auto nav to the landing page even without a logged-in user. I don't believe this is an issue with the App.jsx route changes. I will report this in the Bug Thread.

jackie-leary commented on client/src/components/LandingPage/UserActivity.jsx:
```diff
@@ -0,0 +1,68 @@
+						height: { xs: '220px', sm: '300px' },
+					}}></Paper>
+				<MaterialLink component={Link}>
+					View all currently reading
```
I think the landing page looks great, and I think we should keep the layout you created. Since we've decided to move forward with a single book list that doesn't categorize them, we might want a single link that says "View My Books" instead of the three separate ones. I think we should keep the display of three books that you have, but just grab the most recent additions to the user's book list. We could call the header something like "Recently added to my books". What do you think @jaredsina

jaredsina replied:
I agree, this will allow us to keep most of the work that has already been done on this component. I will move forward with changing it to recently added to my books and update the links!

jackie-leary commented on client/src/components/LandingPage/UserReviewed.jsx:
```diff
@@ -0,0 +1,126 @@
+		review: 'The level of simplicity and user-friendliness in this product has significantly simplified my life. I appreciate the creators for delivering a solution that not only meets but exceeds user expectations.',
+	},
+];
+const UserReviewed = () => {
```
Nitpick: If it's a separate component I think it should have it's own folder, but if it's just data for the landing page we might call it userReviewedLandingPageData instead.

jaredsina replied:
I will move all components to separate folders! I thought since they would only be used on the LandingPage component it would be best to keep them in that folder.

jackie-leary commented on client/src/components/Navbar/Navbar.jsx:
```diff
@@ -8,85 +8,251 @@ import Stack from '@mui/material/Stack';
+const Navbar = () => {
+	const [anchorEl, setAnchorEl] = useState(null);
+	// Needs to be changed for getting if a user is logged in or not
+	const isAuthenticated = true;
```
Are you saying we need to grab isAuthenticated from the store? I think either you or Alex were able to do this for the Login page using 
`const user = useSelector(state => state.session.userInfo);`
<img width="896" alt="image" src="https://github.com/chingu-voyages/v49-tier3-team-29/assets/72801181/7b591b6d-2ef9-42b1-aab1-2723a3de9ea2">

jaredsina replied:
Correct, this functionality was added after this PR was made. I think this should be taken care of in a new PR. Let me know what you think

jackie-leary replied:
Sounds good to me

jaredsina commented:
> Overall everything looks great!
> 
> I did notice a few things, which I have highlighted below:
> 
> * In the NavBar component [Logged In user]: The drop-down menu should automatically close when a click is triggered outside of the menu. See gif below for the current effect when the screen is manually resized.
> 
@alexh205 
The drop-down menu does close when you click outside the menu. In the case that you showed you are clicking outside the site so the menu does not know to close. Clicking anywhere within the site will close the drop-down menu.

jaredsina commented:
@alexh205 Changes were made for the number of books for each screen size. Can be seen here:


https://github.com/chingu-voyages/v49-tier3-team-29/assets/48105264/8a0b9d32-30bb-4b8d-bf76-3248d792b155

jaredsina commented:
@jackie-leary Made the suggested updates! Let me know what you think!

Changes made:
- Moved all components to separate folders
- Updated the recent activity component to recent books component
  - This component was created to display the three recent books that the user added to the list
- Updated this feature branch with the latest changes from the dev branch


https://github.com/chingu-voyages/v49-tier3-team-29/assets/48105264/66e02fe4-9302-4617-8919-fef0781d4e29

alexh205 reviewed (commented):
Thank you for making the updates to the landing page! I have no further comments

jackie-leary reviewed (approved):
Looks great, thank you!
